frontend/class-frontend.php

```diff
@@ -1431,9 +1430,9 @@ function archive_redirect() {
 		global $wp_query;
 
 		if (
-			( $this->options['disable-date'] === true && $wp_query->is_date ) ||
-			( $this->options['disable-author'] === true && $wp_query->is_author ) ||
-			( isset( $this->options['disable-post_formats'] ) && $this->options['disable-post_formats'] && $wp_query->is_tax( 'post_format' ) )
+			$this->options['disable-date'] === true && $wp_query->is_date ||
+			$this->options['disable-author'] === true && $wp_query->is_author ||
+			$wp_query->is_tax( 'post_format' )
```

Comment:
In this case there might actually be a change in behaviour, since if you're diagnosis is correct, this part of the condition was never true before and could be true now.